server.js

```diff
@@ -242,15 +258,50 @@ function render_error(res, user, error) {
 }
 
 // Obtain email parameter from JWT access_token of user
-async function evaluate_user() {
-    var user = null;
-    if (user_access_token) {
-	var decoded = jwt(user_access_token)
-	user = decoded['email'];
-    } 
-    return user;
+async function evaluate_user(req_session) {
+    if (req_session.access_token) {
+		var decoded = jwt(req_session.access_token)
+		if (decoded['email']) {
+			// plain oidc
+			return decoded['email']
+		} else if (decoded['verifiableCredential']){
+			// we have a vc
+			return decoded['verifiableCredential']['credentialSubject']['firstName'] + " "+ decoded['verifiableCredential']['credentialSubject']['familyName']
+		}
+
+	} 
+    return null;
 }
 
+// Get SIOP flow QR code for login via mobile
+function get_siop_qr(req) {
+    let state = req.sessionID
+
+    // Get redirect URI and DID
+    const redirect_uri = config.siop.redirect_uri;
+    const did = config.siop.did;
+
+    // Further parameters
+    const scope = config.siop.scope;
+    const response_type = "vp_token";
+    const response_mode = "post";
+
+    // Build auth request
+    let auth_request = "openid://?";
+    auth_request += "scope="+scope;
+    auth_request += "&response_type="+response_type;
+    auth_request += "&response_mode="+response_mode;
+    auth_request += "&client_id="+did;
+    auth_request += "&redirect_uri="+redirect_uri;
+    auth_request += "&state="+state;
+    auth_request += "&nonce="+crypto.randomBytes(16).toString('base64');
+
+	console.log(auth_request)
```

Comment:
This can be removed or changed to debug(...)

Reply:
true